mcgill-augmented-reality-backend/src/test/java/mcgillar/backend/user/TestUserServiceSignUp.java

```diff
@@ -0,0 +1,227 @@
+package mcgillar.backend.user;
+
+import mcgillar.backend.model.user.AppUser;
+import mcgillar.backend.repositories.AppUserRepository;
+import mcgillar.backend.services.AppUserService;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.invocation.InvocationOnMock;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.any;
+
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
+
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.stubbing.Answer;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.ArrayList;
+
+
+@ExtendWith(MockitoExtension.class)
+public class TestUserServiceSignUp {
+    @Mock
+    private AppUserRepository appUserRepository;
+
+    @Mock
+    PasswordEncoder passwordEncoder;
+    @InjectMocks
+    private AppUserService appUserService;
+    private static final String USERNAME_KEY = "tester";
+    private static final String PASSWORD = "Password123!";
+
+//    private String EMPLOYEE_USERNAME = "TEST_USERNAME";
+//    private String EMPLOYEE_EMAIL = "TEST_EMAIL@mail.ca";
+//    private String EMPLOYEE_PASSWORD = "TEST_PASSWORD";
+//    private String EMPLOYEE_ADDRESS = "TEST_ADDRESS";
+    @BeforeEach
+    public void setMockOutput() {
+        lenient().when(passwordEncoder.encode(PASSWORD)).thenReturn("hashedPassword");
+
+        Answer<?> returnParameterAsAnswer = (InvocationOnMock invocation) -> {
+            return invocation.getArgument(0);
+        };
+        lenient().when(appUserRepository.save(any(AppUser.class))).thenAnswer(returnParameterAsAnswer);
+
+    }
+    @Test
+    public void testCreateUser() {
+
+        AppUser user = null;
+        try {
+            user = appUserService.createUser(USERNAME_KEY, PASSWORD);
+        } catch (IllegalArgumentException e) {
+            // Check that no error occurred
+            fail();
+        }
+        assertNotNull(user);
+    }
+    @Test
+    public void testCreateUserNullUsername() {
+
+        AppUser user = null;
+        try {
+            user = appUserService.createUser(null, PASSWORD);
+        } catch (IllegalArgumentException e) {
+            // Check that no error occurred
+            fail();
```

Comment:
I'm confused, this is the same as the next one except no error message, is that on purpose